tree/ntuple/v7/doc/specifications.md

```diff
@@ -438,14 +438,22 @@ Top-level fields have their own field ID set as parent ID.
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 +                            Field ID                           +
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                             Flags                             |
+|             Flags             |      Representation Index     |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 ```
 
 The order of columns matter: every column gets an implicit column ID
 which is equal to the zero-based index of the column in the serialized list.
 Multiple columns attached to the same field should be attached from smaller to larger IDs.
 
+A field can have multiple alternative column representations.
```

Comment:
I think we should already mention up here the constraints on multiple column representations. We may also have a terminology problem regarding "cardinality": as far as I understand, please correct if wrong, we expect all column representations to have the same number of columns *and* the same number of elements in each of them.

For example, UTF-8 and UTF-16 encoded strings stored as code points are fine, but each UTF-16 code point stored as two characters is not because it results in double the number of elements...